common/src/main/java/whocraft/tardis_refined/common/blockentity/device/AstralManipulatorBlockEntity.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    public void sendUpdates() {
+        level.sendBlockUpdated(this.getBlockPos(), level.getBlockState(this.getBlockPos()), level.getBlockState(this.getBlockPos()), Block.UPDATE_CLIENTS);
+        setChanged();
+    }
+
+    public CompoundTag getUpdateTag() {
+        CompoundTag tag = new CompoundTag();
+        saveAdditional(tag);
+        return tag;
+    }
+
+
+    private void attemptToBuild(BlockPos pointA, BlockPos pointB) {
+
+        var submittedBlocks = new ArrayList<ManipulatorCraftingRecipeItem>();
+
+        // Grab the absolutes for these items.
+        var xDiff = Math.abs(pointA.getX() - pointB.getX());
+        var yDiff = Math.abs(pointA.getY() - pointB.getY());
+        var zDiff = Math.abs(pointA.getZ() - pointB.getZ());
+
+        var smallestPointX = pointA.getX() > pointB.getX() ? pointB.getX() : pointA.getX();
+        var smallestPointY = pointA.getY() > pointB.getY() ? pointB.getY() : pointA.getY();
+        var smallestPointZ = pointA.getZ() > pointB.getZ() ? pointB.getZ() : pointA.getZ();
+
+        var minPoint = new BlockPos(smallestPointX, smallestPointY, smallestPointZ);
+
+        for (int y = 0; y < yDiff + 1; y++) {
+
+            for (int x = 0; x < xDiff + 1; x++) {
+
+                for (int z = 0; z < zDiff + 1; z++) {
+
+                    var blockPosInWorld = new BlockPos(minPoint.getX() + x, minPoint.getY() + y, minPoint.getZ() + z);
+
+                    submittedBlocks.add(new ManipulatorCraftingRecipeItem(new BlockPos(x, y, z), level.getBlockState(blockPosInWorld).getBlock()));
+                }
+            }
+        }
+
+        // Filter recipes by the first block, will make it a LOT easier later down the line. (when I make it)
+        var firstBlock = submittedBlocks.stream().filter(x -> x.relativeBlockPos.getX() == 0 && x.relativeBlockPos.getY() == 0 && x.relativeBlockPos.getZ() == 0).findFirst();
+        if (!firstBlock.isPresent()) {
+            return;
+        }
+
+        var possibleRecipes = new ArrayList<ManipulatorCraftingRecipe>();
+
+        for (var recipe : ManipulatorCrafting.MANIPULATOR_CRAFTING_RECIPES) {
+
+            var zeroPos = recipe.itemList.stream().filter(x -> x.relativeBlockPos.getX() == 0 && x.relativeBlockPos.getY() == 0 && x.relativeBlockPos.getZ() == 0).findFirst();
+            if (zeroPos.isPresent()) {
+                if (zeroPos.get().block == firstBlock.get().block) {
+                    possibleRecipes.add(recipe);
+                }
+            }
+
+        }
+
+        // No recipes, so no result!
+        if (possibleRecipes.size() == 0) {
+            return;
+        }
+
+        for (var recipe : possibleRecipes) {
+
+            if (recipe.hasSameItems(submittedBlocks)) {
+
+                // Remove the recipe blocks.
+                for (var blockPos : BlockPos.betweenClosed(pointA, pointB)) {
+                    level.setBlock(blockPos, Blocks.AIR.defaultBlockState(), Block.UPDATE_ALL);
```

Comment:
I think level.removeBlock may suffice better?

Reply:
Using `level.removeBlock` this way sends a block update to the blocks around including ones in the crafting. Buttons, and any other block that cannot exist without sitting on another block will drop as an item.